mono/mini/interp/transform.c

```diff
@@ -4758,24 +4798,10 @@ mono_interp_transform_method (InterpMethod *imethod, ThreadContext *context, Mon
 					char *wrapper_name = g_strdup_printf ("__icall_wrapper_%s", mi->name);
 					nm = mono_marshal_get_icall_wrapper (mi->sig, wrapper_name, mi->func, TRUE);
```

Comment:
why do we still need the logic for the `ctor` here?

Reply:
the ctor calls are not handled by this PR (the code that I added was in g_assert_not_reached for debugging). This is because, for now, we call ctors directly from intep (in NEWOBJ) instead of transforming the call